src/timeline/TimelineModel.cpp

```diff
@@ -773,6 +773,16 @@ TimelineModel::formatDateSeparator(QDate date) const
         return date.toString(fmt);
 }
 
+QString
+TimelineModel::getMessageFromId(QString id) const
+{
+    auto e = events.get(id.toStdString(), "", false);
+    if (!e)
+        return "";
+    std::string body = mtx::accessors::body(*e);
+    return QString::fromStdString(body);
+}
+
```

Comment:
We already have getDump(eventId), which you can call and then access body and such as you like in Qml.